src/scripts/convert_lovd2.php

```diff
@@ -0,0 +1,1318 @@
+    lovd_setUserIDSettings($_POST['submitterid_fixed'], $_POST['submitterid_translation'],
+        $_POST['curatorid_fixed'], $_POST['curatorid_translation']);
+
+    return !lovd_error();
+}
+
+
+
+
+
+function main ($aFieldLinks, $aSections, $aCustomColLinks)
+{
+    global $_DB, $_T, $_SETT, $_WARNINGS;
+
+    // Only allow managers or higher to perform conversion.
+    lovd_requireAUTH(LEVEL_MANAGER);
+
+    // Selected transcript, will be created from request arguments.
+    $zTranscript = null;
+
+    // Determine max file upload size in bytes.
+    $nMaxSizeLOVD = 100*1024*1024; // 100MB LOVD limit. Note: value copied from import.php
+    $nMaxSize = min($nMaxSizeLOVD,
+        lovd_convertIniValueToBytes(ini_get('upload_max_filesize')),
+        lovd_convertIniValueToBytes(ini_get('post_max_size')));
+
+    // Print header
+    define('PAGE_TITLE', 'Convert LOVD2 export to LOVD3 import');
+    $_T->printHeader(false);
+    $_T->printTitle();
+
+    if (!POST) {
+        // Show file upload form.
+        lovd_showConversionForm($nMaxSizeLOVD, $nMaxSize);
+        return;
+    } else {
+        if (!empty($_POST['transcriptid'])) {
+            $qTranscript = $_DB->query('SELECT t.*, chromosome FROM ' . TABLE_TRANSCRIPTS .
+                ' AS t LEFT JOIN ' . TABLE_GENES . ' AS g ON (g.id = t.geneid)  WHERE t.id = ?',
+                array($_POST['transcriptid']));
+            $zTranscript = $qTranscript->fetchAssoc();
+        }
+        if (!lovd_validateConversionForm($zTranscript, $nMaxSize,
+            $nMaxSizeLOVD)) {
+            lovd_showConversionForm($nMaxSizeLOVD, $nMaxSize);
+            return;
+        }
+    }
+
+    $oProgressBar = new ProgressBar('', 'Parsing file...', 'Done.');
+
+    $aData = lovd_openLOVD2ExportFile($_POST, $_FILES);
+
+    // Get headers for input (as defined in file) and output (per section).
+    list($aInputHeaders, $aOutputHeaders) = lovd_getHeaders($aData, $aFieldLinks,
+        $aSections, $aCustomColLinks);
+
+    if ($aData === false || $aInputHeaders === false) {
+        lovd_showConversionForm($nMaxSizeLOVD, $nMaxSize);
+        return;
+    }
+
+    // Parse data and get output records per section.
+    list($aOut['vog'],
+        $aOut['vot'],
+        $aOut['disease'],
+        $aOut['individual'],
+        $aOut['i2d'],
+        $aOut['screening'],
+        $aOut['s2g'],
+        $aOut['s2v'],
+        $aOut['phenotype']) = lovd_parseData($aData, $zTranscript, $aFieldLinks, $aInputHeaders,
+        $aOutputHeaders, $aSections, $oProgressBar);
+
+    print('<H3>Conversion log:</H3>
+    <TEXTAREA id="header_log" cols="100" rows="10" style="font-family: monospace; 
+        white-space: nowrap; overflow: scroll;">' .
+        implode("\n", $_WARNINGS) .
+    '</TEXTAREA><BR><BR>');
+
+    if (lovd_error()) {
+        print('<B>There were fatal errors during conversion:</B>');
+        lovd_errorPrint();
+    }
```

Comment:
> I think you can't get here if you have lovd_error() returning true. I can't be sure, as lovd_errorAdd() is called in 5 different functions, and there's no way for me to see in a structured way how they are calling each other.

Do you mean you can't see a call to `lovd_errorAdd()` that would trigger this print statement? `lovd_getDiseaseID()` can add errors during conversion if the search for a disease name is ambiguous.

> If you can't get here, this part can go. If you can get here, the bottom part shouldn't be displayed if lovd_errorAdd() is true.

I thought the conversion output might be useful somehow. What else would you want displayed, the submission form?